### PR TITLE
Style Route Planner

### PR DESCRIPTION
**Pull request by AconexOfficial**
### Description

Style the route planner component aswell as the journey page as a whole according to wireframes and discussions using Ionic and CSS.

## Checklist

- [x] My code follows the style guidelines
- [x] I have documented my code
- [x] My changes generate no new warnings
- [x] I have linked the Issue
- [x] The Pull Request covers all the Checkpoints from the tagged Issue
- [x] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [x] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [x] No tests required
- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by Paulomart** [comment]
@AconexOfficial When adding so much css please consider using css modules: https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/

File: frontend/src/components/Icons.module.css [hunk 1]
```diff
@@ -0,0 +1,45 @@
+.button_primary {
+background-color: ion-color-primary;
```
Comment (timvahlbrock) [inline frontend/src/components/Icons.module.css:2]: wrong indentation

Comment (timvahlbrock) [reply frontend/src/components/Icons.module.css:2]: and why is button-styling for non-icon buttons within the Icons.module.css?

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css [hunk 2]
```diff
@@ -0,0 +1,93 @@
+ion-modal#favorite_dialogue {
```
Comment (timvahlbrock) [inline frontend/src/components/RoutePlanner/RoutePlanner.module.css:1]: Why is this styling specific for the favourite dialogue? Wasn't the goal to style the components in a reusable way? And why do the modals require height configuration at all? Shouldn't the height be determined automatically by the size of the content?

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  --height: 25%;
+  --width: 90%;
+  --border-radius: 16px;
+  --box-shadow: 0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1);
```
Comment (timvahlbrock) [inline frontend/src/components/RoutePlanner/RoutePlanner.module.css:5]: Please don't use pixels as a unit. This makes the application unresponsive.

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+ion-modal#favorite_dialogue::part(backdrop) {
+  background: rgba(209, 213, 219);
```
Comment (timvahlbrock) [inline frontend/src/components/RoutePlanner/RoutePlanner.module.css:9]: why is there an abitrary color used here? Use color variables, ideally existing ones (ion-color-primary, etc.)

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+.toggle_button_label{
+  align-self: center;
+}
+
+.date_time_button {
+  color: primary;
+}
+
+.date_icon {
+  font-size: 4vh;
```
Comment (timvahlbrock) [inline frontend/src/components/RoutePlanner/RoutePlanner.module.css:68]: don't use vh as a font-size. use rem.

Comment (timvahlbrock) [reply frontend/src/components/RoutePlanner/RoutePlanner.module.css:68]: I'm not even sure vh is working with fonts.

File: frontend/src/pages/JourneyPage.tsx [hunk 6]
```diff
@@ -27,14 +27,16 @@ const JourneyPage: React.FC = () => {
   return (
     <IonPage id="main-content">
       <IonHeader>
-        <IonToolbar>
+        <IonToolbar className="menuBar">
```
Comment (timvahlbrock) [inline frontend/src/pages/JourneyPage.tsx:30]: I think this class name name might be redundant now.

**Review by timvahlbrock — CHANGES_REQUESTED** [review]
In my opinion this needs some more work. But maybe @Paulomart has another opinion, I haven't been in touch a lot with the styling of the app.

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css (hunk 5 above)
Comment (AconexOfficial) [reply frontend/src/components/RoutePlanner/RoutePlanner.module.css:68]: It does work, but changed it to rem.

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css (hunk 4 above)
Comment (AconexOfficial) [reply frontend/src/components/RoutePlanner/RoutePlanner.module.css:9]: Was adoped from existing route planner CSS. Will rework to use classes and variables

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css (hunk 3 above)
Comment (AconexOfficial) [reply frontend/src/components/RoutePlanner/RoutePlanner.module.css:5]: Was adoped from existing route planner CSS. Will rework to use relative sizes

File: frontend/src/components/RoutePlanner/RoutePlanner.module.css (hunk 2 above)
Comment (AconexOfficial) [reply frontend/src/components/RoutePlanner/RoutePlanner.module.css:1]: Was adoped from existing route planner CSS. Will rework to be reusable as a class

File: frontend/src/components/Icons.module.css (hunk 1 above)
Comment (AconexOfficial) [reply frontend/src/components/Icons.module.css:2]: You're right, now that I think about it. Renamed to Buttons.module.css

**Comment by Paulomart** [comment]
A few general comments: 

* I would suggest using one css file for each component
* Please dont use id in css.
* ####  _Please **dont** use id in css._
* ### Please dont **use** id in css.
* **Please dont use id **in** css.**
* ## Please dont use id in css.
* __**Please dont use id in css.**__
* # Please dont use id in css.

**Comment by AconexOfficial** [comment]
> A few general comments:
> 
> * I would suggest using one css file for each component
> * Please dont use id in css.
> * #### _Please **dont** use id in css._
> * ### Please dont **use** id in css.
> * **Please dont use id in css.**
> * ## Please dont use id in css.
> * **Please dont use id in css.**
> * # Please dont use id in css.

I just moved the stuff with the ids over from the old RoutePlanner.css to the module because I couldn't check what exactly those were there for. Already changing them into regular classes.

Also regarding the one module per component: I thought maybe some resuable stuff like button styles would have made sense in a separate module.css to be reused somewhere else. Can possibly merge the modules though.

File: frontend/src/components/Icons.module.css (hunk 1 above)
Comment (AconexOfficial) [reply frontend/src/components/Icons.module.css]: Moved all CSS into RoutePlanner.module.css and completely removed the Icons/Buttons.module.css

**Comment by Paulomart** [comment]
> Also regarding the one module per component: I thought maybe some resuable stuff like button styles would have made sense in a separate module.css to be reused somewhere else. Can merge the modules though.

Totally agree here. But maybe thoses 'resuable stuff' should then be components then, each with its own js and css file.

**Comment by AconexOfficial** [comment]
> > Also regarding the one module per component: I thought maybe some resuable stuff like button styles would have made sense in a separate module.css to be reused somewhere else. Can merge the modules though.
> 
> Totally agree here. But maybe thoses 'resuable stuff' should then be components then, each with its own js and css file.

For now to keep things going I merged all css into RoutePlanner.module.css then.

**Review by Paulomart — APPROVED** [review]

**Review by timvahlbrock — APPROVED** [review]

**Pull request merged**